test/include/util/storage_test_util.h

```diff
@@ -170,63 +190,40 @@ struct StorageTestUtil {
       }
     }
   }
-};
 
-// This does NOT return a sensible tuple in general. This is just some filler
-// to write into the storage layer and is devoid of meaning outside of this class.
-struct FakeRawTuple {
-  template <typename Random>
-  FakeRawTuple(const storage::BlockLayout &layout, Random *generator);
-  ~FakeRawTuple() { delete[] contents_; }
 
-  // Since all fields we store in pages are equal to or shorter than 8 bytes,
-  // we can do equality checks on uint64_t always.
-  // 0 return for non-primary key indexes should be treated as null.
-  uint64_t Attribute(const storage::BlockLayout &layout, uint16_t col) const {
-    return storage::StorageUtil::ReadBytes(layout.attr_sizes_[col], contents_ + attr_offsets_[col]);
-  }
-
-  const storage::BlockLayout &layout_;
-  std::vector<uint32_t> attr_offsets_;
-  byte *contents_;
-};
-
-struct TupleAccessStrategyTestUtil {
-  TupleAccessStrategyTestUtil() = delete;
-  // Fill the given location with the specified amount of random bytes, using the
-  // given generator as a source of randomness.
-  template <typename Random>
-  static void FillWithRandomBytes(uint32_t num_bytes, byte *out, Random *generator) {
-    std::uniform_int_distribution<uint8_t> dist(0, UINT8_MAX);
-    for (uint32_t i = 0; i < num_bytes; i++) out[i] = static_cast<byte>(dist(*generator));
-  }
-
-  // Write the given fake tuple into a block using the given access strategy,
+  // Write the given tuple (projected row) into a block using the given access strategy,
   // at the specified offset
-  static void InsertTuple(const FakeRawTuple &tuple, const storage::TupleAccessStrategy *tested,
+  static void InsertTuple(const storage::ProjectedRow *tuple, const storage::TupleAccessStrategy *tested,
                           const storage::BlockLayout &layout, const storage::TupleSlot slot) {
-    for (uint16_t col = 0; col < layout.num_cols_; col++) {
-      uint64_t col_val = tuple.Attribute(layout, col);
-      if (col_val != 0 || col == PRESENCE_COLUMN_ID)
-        storage::StorageUtil::WriteBytes(layout.attr_sizes_[col], tuple.Attribute(layout, col),
-                                         tested->AccessForceNotNull(slot, col));
-      else
+    // Skip the version vector for tuples
+    for (uint16_t col = 1; col < layout.num_cols_  ; col++) {
+      const byte *val_ptr = tuple->AccessWithNullCheck(uint16_t(col - 1));
+      if (val_ptr == nullptr) {
         tested->SetNull(slot, col);
-      // Otherwise leave the field as null.
+      } else {
+        // Read the value and convert it to uint64_t
+        uint64_t val = 0;
```

Comment:
The documentation for AccessWithNullCheck suggests that we "reinterpret_cast and dereference to access the value", I think this is preferable to a memcpy.